### PR TITLE
fix/clippy: fix clippy warnings

### PR DESCRIPTION
**Pull request by nbaksalyar**

### CODE REVIEW

File: src/rate_limiter.rs [hunk 1]
```diff
@@ -248,7 +248,7 @@ impl RateLimiter {
             None => return None,
         };
 
-        let deduction = amount_charged.saturating_sub(part_count as u64 * MAX_PART_LEN as u64);
+        let deduction = amount_charged.saturating_sub(u64::from(part_count) * MAX_PART_LEN as u64);
```
Comment (madadam) [inline src/rate_limiter.rs]: Shouldn't the `MAX_PART_LEN as u64` be also rewritten to `u64::from(MAX_PART_LEN)`?

File: src/states/peer.rs [hunk 2]
```diff
@@ -180,7 +180,7 @@ impl Peer {
 
     #[cfg_attr(feature = "cargo-clippy", allow(too_many_arguments))]
     pub fn from_bootstrapping(
-        our_section: (Prefix, BTreeSet<PublicId>),
+        our_section: &(Prefix, BTreeSet<PublicId>),
```
Comment (madadam) [inline src/states/peer.rs]: Might be better to change this to `our_section: (Prefix, &BTreeSet<PublicId>)`, so the called doesn't have to clone the `BTreeSet` when passing it in, as it doesn't seem to be consumed inside this function.

File: src/rate_limiter.rs (hunk 1 above)
Comment (maqi) [reply src/rate_limiter.rs]: MAX_PART_LEN  is usize   and `u64::from(usize)` is not implemented  :(

File: src/timer.rs [hunk 3]
```diff
@@ -239,6 +239,10 @@ mod implementation {
 
 #[cfg(feature = "use-mock-crust")]
 mod implementation {
+    // This lint requires to make changes that might introduce discrepancies with the actual Crust code,
+    // so we turn it off for mock crust specifically.
+    #![cfg_attr(feature = "cargo-clippy", allow(needless_pass_by_value))]
```
Comment (maqi) [inline src/timer.rs]: will it be better to put this into lib.rs,  just as what the current master branch do?

File: src/rate_limiter.rs (hunk 1 above)
Comment (madadam) [reply src/rate_limiter.rs]: Got it. All good then.

**Pull request merged**